ledger_secure_sdk_sys/build.rs

```diff
@@ -71,48 +71,212 @@ const CCID_FILES: [&str; 9] = [
+    };
+    (c_sdk_hash, c_sdk_version)
+}
+
+fn retrieve_makefile_infos(bolos_sdk: &Path) -> Result<(u32, String), SDKBuildError> {
+    let makefile_defines =
+        File::open(bolos_sdk.join("Makefile.defines")).expect("Could not find Makefile.defines");
+    let mut api_level: i32 = -1;
+    let mut sdk_name = String::new();
+    for line in BufReader::new(makefile_defines).lines().flatten() {
+        if line.contains("API_LEVEL") {
+            api_level = line.split(":=").collect::<Vec<&str>>()[1]
+                .trim()
+                .parse()
+                .map_err(|_| SDKBuildError::InvalidAPILevel)?;
+        } else if line.contains("SDK_NAME") {
+            sdk_name = line.split(":=").collect::<Vec<&str>>()[1]
+                .trim()
+                .to_string();
+        }
+    }
+
+    if sdk_name.is_empty() || api_level == -1 {
+        return Err(SDKBuildError::CouldNotGetMakefileInfos);
+    } else {
+        return Ok((api_level as u32, sdk_name));
+    }
+}
+
+fn retrieve_target_file_infos(
+    device: &Device,
+    bolos_sdk: &Path,
+) -> Result<(String, String), SDKBuildError> {
+    let prefix = if *device == Device::NanoS {
+        "".to_string()
+    } else {
+        format!("target/{}/", device.to_string())
+    };
+    let target_file_path = bolos_sdk.join(format!("{}include/bolos_target.h", prefix));
+    println!("cargo:warning=Target file path is {:?}", target_file_path);
+    let target_file =
+        File::open(target_file_path).map_err(|_| SDKBuildError::TargetFileNotFound)?;
+    let mut target_id = String::new();
+    let mut target_name = String::new();
+    for line in BufReader::new(target_file).lines().flatten() {
+        if line.contains("TARGET_ID") {
+            let id = line.split(" ").collect::<Vec<&str>>()[2];
+            if id.starts_with("0x") {
+                target_id = id.to_string();
+            }
+        } else if line.contains("TARGET_") {
+            target_name = line.split(" ").collect::<Vec<&str>>()[1].to_string();
+        }
+    }
+    if target_id.is_empty() || target_name.is_empty() {
+        return Err(SDKBuildError::CouldNotGetTargetFileInfos);
+    } else {
+        return Ok((target_id, target_name));
+    }
+}
+
 /// Fetch the appropriate C SDK to build
-fn clone_sdk(device: &Device) -> (PathBuf, u32) {
-    let (repo_url, sdk_branch, api_level) = match device {
-        Device::NanoS => ("https://github.com/LedgerHQ/nanos-secure-sdk", "master", 0),
+fn clone_sdk(device: &Device) -> Result<SDKInfo, SDKBuildError> {
+    let mut sdk_info = SDKInfo::new();
+    let (repo_url, sdk_branch, api_level, name) = match device {
+        Device::NanoS => (
+            "https://github.com/LedgerHQ/nanos-secure-sdk",
+            "master",
+            0,
+            "nanos-secure-sdk",
+        ),
         Device::NanoX => (
             "https://github.com/LedgerHQ/ledger-secure-sdk",
             "API_LEVEL_5",
             5,
+            "ledger-secure-sdk",
```

Comment:
the `name` could be retrieved by making the repo_url a PathBuf and using https://doc.rust-lang.org/std/path/struct.PathBuf.html#method.file_name

Reply:
Ok